azurerm/internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -687,6 +685,7 @@ func resourceArmCosmosDbAccountApiUpsert(client *documentdb.DatabaseAccountsClie
 	stateConf := &resource.StateChangeConf{
 		Pending:    []string{"Creating", "Updating", "Deleting", "Initializing"},
 		Target:     []string{"Succeeded"},
+		Timeout:    180 * time.Minute,
```

Comment:
this should be using the user specified timeout - rather than a hard-coded one

Reply:
updated